conda_env/specs/yaml_file.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+from .. import env
+
+
+class YamlFileSpec(object):
+    def __init__(self, filename=None, name=None, **kwargs):
+        self.filename = filename
+        self.name = name
+
+    def can_handle(self):
+        self._environment = env.from_file(self.filename)
```

Comment:
The problem throwing from inside `can_handle` is that it provides a name that appears to answer a yes/no question.  Thinking in terms of true or false, if I ask if you want to go to lunch I expect you to say yes or no, not yes or flip a table. :-)

Exceptions are for behavior that's exceptional -- a behavior you wouldn't expect.  It's reasonable for `detect` to fail with an exception as you expect `detect` to work, but `can_handle` implies a true/false.

Given your example, it might even be worth adding an `obj.error_message` to each loader so it explains what it couldn't do.  That way:

```python
if not loader.can_handle:
    errors.append(loader.error_message)
    continue
env = loader.detect()
break
```

Then detect for `env is None` and display the error messages if that's the case, otherwise proceed.